invenio_circulation/api.py

```
@@ -159,6 +165,53 @@ def create(cls, data, id_=None):
             data['_circulation']['status'] = ItemStatus.ON_SHELF
         return super(Item, cls).create(data, id_=id_)
 
+    @classmethod
+    def find_by_holding(cls, **kwargs):
+        """Find item versions based on their holdings information.
+
+        Every given kwarg will be queried as a key-value pair in the items
+        holding.
+
+        :returns: List[(UUID, version_id)] with `version_id` as used by
+                  `RecordMetadata.version_id`.
+        """
+        def _get_filter_clause(obj, key, value):
+            val = obj[key].astext
+            CASTS = {
```

Ok, let's raise an exception for 1 and 3+ values for now.